Runtime/Scripts/BaseMeshSync.Textures.cs

```diff
@@ -0,0 +1,210 @@
+                        // If that happens, save it for later and try again:
+                        if (!SetSerializedTextureForMaterial(mat, textureName)) {
+                            pendingMaterialUpdates.Add(new Tuple<Material, string>(mat, textureName));
+
+                            // Set the texture on the material to null for now so the render texture
+                            // is not saved to file again if this gets called again before it was
+                            // set to the texture in the asset library:
+                            mat.SetTexture(textureName, null);
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Sets the texture on the material if it exists in the AssetDatabase, otherwise it schedules a delayed call to try again when the asset might be imported.
+        /// </summary>
+        /// <param name="mat">Material to set the texture on</param>
+        /// <param name="textureName">Name of the texture in the material</param>
+        /// <returns></returns>
+        private bool SetSerializedTextureForMaterial(Material mat, string textureName) {
+            var savePath = GetSavePath(mat, textureName);
+
+            // LoadAssetAtPath can throw an exception if it's called during a domain backup but we have no way to detect that:
+            try {
+                if (AssetDatabase.LoadAssetAtPath<Texture2D>(savePath) == null) {
+                    EditorApplication.delayCall -= UpdatePendingMaterials;
+                    EditorApplication.delayCall += UpdatePendingMaterials;
+
+                    return false;
+                }
+            }
+            catch (Exception ex) {
+                return false;
+            }
+
+            // Ensure the texture is up to date:
+            AssetDatabase.ImportAsset(savePath);
+
+            var savedTexture = AssetDatabase.LoadAssetAtPath<Texture2D>(savePath);
+
+            mat.SetTexture(textureName, savedTexture);
+
+            if (textureName == MeshSyncConstants._MetallicGlossMap ||
+                textureName == MeshSyncConstants._MaskMap) {
+                mat.EnableKeyword(MeshSyncConstants._METALLICGLOSSMAP);
+                mat.EnableKeyword(MeshSyncConstants._METALLICSPECGLOSSMAP);
+
+                TextureImporter importer = (TextureImporter)AssetImporter.GetAtPath(savePath);
+                if (importer != null) {
+                    importer.sRGBTexture = false;
+                    importer.alphaIsTransparency = false;
+                    importer.ignorePngGamma = true;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Tries to apply any pending material updates and unschedules itself when done.
+        /// </summary>
+        private void UpdatePendingMaterials() {
+            EditorApplication.delayCall -= UpdatePendingMaterials;
```

Comment:
Do we need to unregister? from the docs, I understand that this will be called only once after being added. https://docs.unity3d.com/ScriptReference/EditorApplication-delayCall.html

Reply:
A comment when you register the function might be better. If I am reading this code I would have to stop to understand why we are unregistering. We should be eliminating dead code when we can

Reply:
I've removed it now